src/utility/Errors.ts

```diff
@@ -20,9 +20,9 @@ export class FileNotFoundError extends Error {
 }
 
 export class RetriesExceededError extends Error {
-    public constructor(error: Error) {
-        super(`Maximum number of retries exceeded. The operation failed with error: ${error.message}.`);
+    public constructor(error: Error, operationName: string) {
+        super(`Maximum number of retries exceeded. The operation '${operationName}' failed with error: ${error.message}. Technical details: ${JSON.stringify(error)}`);
```

Comment:
Could you please confirm this works in our case? I remember that when I tried to stringify an `error` in the past, it failed because of a circular object issue. It also seems that some people mention that stringifying errors result in empty objects (https://levelup.gitconnected.com/beware-of-using-json-stringify-for-logging-933f18626d51).